ocpmodels/datasets/oc22_lmdb_dataset.py

```diff
@@ -32,6 +32,11 @@ class OC22LmdbDataset(Dataset):
     Useful for Structure to Energy & Force (S2EF), Initial State to
     Relaxed State (IS2RS), and Initial State to Relaxed Energy (IS2RE) tasks.
 
+    The keys in the LMDB must be integers (stored as ascii objects) starting
+    from 0 through the length of the LMDB. For historical reasons any key names
+    "length" is ignored since that was used to infer length of many lmdbs in the same
+    folder, but lmdb lengths are now calculated directly from the number of keys.
+
```

Comment:
Given how similar this class is to `LmdbDataset`, we should figure out a way to avoid repetitive code. This does not have to be in this PR, but something to keep in mind.

Reply:
I agree - I don't understand precisely why these were different, but we should clean this up after.